### ISSUE TITLE
Question:  Proper way to add custom helper functions?

### ISSUE THREAD

**Issue by anakinjay** [zfcampus/zf-apigility#139]
New to Apigility and Zend framework.  We're moving from a custom in house built API to apigility, and I'm stumped how to accomplish this.

Basically we have some developers working on macs, and some on windows...  we push to a windows server, and query off a MSSQL database.  The code for connecting to a MSSQL database is different for macs than windows because you have to use a different driver.

We get around this by using a custom DB connection helper which detects the server OS and uses the proper driver.  So someone can develop locally on a mac and when they push to the windows server, the helper knows to switch drivers.

So this helper needs to be available to all APIs created in apigility.  I realize I could probably define it in one of the global configuration files, but that seems pretty hacky.

Any advice on where to start looking for the proper way to provide a global helper function?

Thanks guys!

**Comment by TomHAnderson** [comment]
Allow me to blow your development world open:  Vagrant
Your question is really outside the scope of Apigility and falls squarely on the dev-ops in support of your application.  You need to change your development ways so all development is done inside a Vagrant virtual machine so all developers regardless of OS will have the same *nix-based environment running on their machine which is as close as possible to the environment of deployment.

You may start here:  https://www.puphpet.com

**Comment by TomHAnderson** [comment]
And/or you may create a virtual machine running Windows on Vagrant.

**Comment by anakinjay** [comment]
I've used vagrant before when developing on windows machines to push to nix servers... the problem here is we're pushing to a windows server, so we'd have to do a full install of windows server inside of vagrant in order to make that work which seems like WAY more overhead than just building a simple helper function to switch drivers.

That being said, I only described the problem for context.  My real question is what's the proper way to create a class or function that is available to access from all apigility controllers.
